### PR TITLE
Changes in opm-common for current Production/Injection Group Control parameters

### PR DESCRIPTION
**Pull request by jalvestad**
This pullrequest includes changes in opm-common needed to hold the current Production and Injection Group control parameters for use in output to summary file and for use in output of Eclipse Compatible restart file. The changes to write the control status to the summary file is also included. 

This pull request is needed for opm/opm-simulators#2459 to be able to build.

The pull request is ready for review and merge from my side.

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

File: src/opm/output/eclipse/Summary.cpp [hunk 1]
```diff
@@ -688,6 +742,70 @@ inline quantity potential_rate( const fn_args& args ) {
     return { sum, rate_unit< phase >() };
 }
 
+template < bool isGroup = true, bool Producer = true, bool waterInjector = false, bool gasInjector = false>
```
Comment (joakim-hove) [inline src/opm/output/eclipse/Summary.cpp]: Is the default initialization used/needed?

**Comment by bska** [comment]
jenkins build this opm-simulators=2459 please

File: src/opm/output/eclipse/Summary.cpp (hunk 1 above)
Comment (jalvestad) [reply src/opm/output/eclipse/Summary.cpp]: I agree, I do not think the default is needed - I will remove it.

**Comment by bska** [comment]
jenkins build this opm-simulators=2459 please

**Comment by bska** [comment]
Thanks a lot for the update.  Commit a4d3ec3 in particular was necessary to get this to run without timeouts on the CI system.  Now the only remaining thing to do is to update the reference data to include new summary vectors in the expected results for [model1/BASE_MODEL_1.DATA](https://github.com/OPM/opm-tests/blob/15cf9b46cf5f84fbdca1a005e21f527eebc82498/model1/BASE_MODEL_1.DATA), [model1/FAULTS_MODEL_1.DATA](https://github.com/OPM/opm-tests/blob/15cf9b46cf5f84fbdca1a005e21f527eebc82498/model1/FAULTS_MODEL_1.DATA), and [model1/MSW_MODEL_1.DATA](https://github.com/OPM/opm-tests/blob/15cf9b46cf5f84fbdca1a005e21f527eebc82498/model1/MSW_MODEL_1.DATA).  Doing so will fix the comparison failures below.

I will however defer the data update until OPM/opm-common#1599 and its associate update OPM/opm-tests#266 have been merged into master.

---

```
Keywords found in second case, but missing in first case: 

  > 'GMCTP:FIELD_N'
  > 'GMCTP:FIELD_S'
  > 'GMCTP:INJ_N'
  > 'GMCTP:INJ_S'
  > 'GMCTP:PROD_N'
  > 'GMCTP:PROD_S'
```

**Comment by bska** [comment]
jenkins build this opm-simulators=2459 opm-tests=267 please

**Comment by bska** [comment]
> jenkins build this opm-simulators=2459 opm-tests=267 please

Success!  Thanks a lot for your patience.  I am going to merge this and the downstream PRs now.

**Pull request merged**